src/service/data_products/samples.py

```diff
@@ -203,71 +204,49 @@ async def get_samples(
     if status_only:
         return _response(dp_match=dp_match, dp_sel=dp_sel)
-    if count:
-        # for now this method doesn't do much. One we have some filtering implemented
-        # it'll need to take that into account.
-        # may want to make some sort of shared builder
-        count = await count_simple_collection_list(
-            appstate.arangostorage,
-            names.COLL_SAMPLES,
-            collection_id,
-            load_ver,
-            match_spec=match_spec,
-            selection_spec=selection_spec,
-        )
-        return _response(dp_match=dp_match, dp_sel=dp_sel, count=count)
-    else:
-        res = await query_table(
-            appstate.arangostorage,
-            names.COLL_SAMPLES,
-            collection_id,
-            load_ver,
-            sort_on,
-            sort_descending=sort_desc,
-            skip=skip,
-            limit=limit,
-            output_table=output_table,
-            match_spec=match_spec,
-            selection_spec=selection_spec,
-            document_mutator=_remove_keys,
-        )
-        return _response(
-            skip=res.skip,
-            limit=res.limit,
-            dp_match=dp_match,
-            dp_sel=dp_sel,
-            fields=res.fields,
-            table=res.table,
-            data=res.data
-        )
+    filters = FilterSet(
+        collection_id,
+        load_ver,
+        collection=names.COLL_SAMPLES,
+        count=count,
+        match_spec=SubsetSpecification(
+            subset_process=dp_match, mark_only=match_mark, prefix=MATCH_ID_PREFIX),
+        selection_spec=SubsetSpecification(
+            subset_process=dp_sel, mark_only=selection_mark, prefix=SELECTION_ID_PREFIX),
+        sort_on=sort_on,
+        sort_descending=sort_desc,
+        skip=skip,
+        limit=limit,
+    )
+    res = await query_table(appstate.arangostorage, filters, output_table, _remove_keys)
+    return _response(dp_match=dp_match, dp_sel=dp_sel, res=res)
 
 
 def _response(
     dp_match: models.DataProductProcess = None,
     dp_sel: models.DataProductProcess = None,
-    count: int = None,
-    skip: int = 0,
-    limit: int = 0,
-    fields: list[dict[str, str]] = None,
-    table: list[list[Any]] = None,
-    data: list[dict[str, Any]] = None,
+    res: QueryTableResult = None,
 ) -> SamplesTable:
-    return SamplesTable(
-        skip=skip,
-        limit=limit,
-        count=count,
-        match_state=dp_match.state if dp_match else None,
-        selection_state=dp_sel.state if dp_sel else None,
-        fields=fields,
-        table=table,
-        data=data,
-    )
+    if res:
+        return SamplesTable(
+            skip=res.skip,
+            limit=res.limit,
+            count=res.count,
+            match_state=dp_match.state if dp_match else None,
+            selection_state=dp_sel.state if dp_sel else None,
+            fields=res.fields,
+            table=res.table,
+            data=res.data,
+        )
+    else:
```

Comment:
I would personally avoid `else` here.

Reply:
Here we have a difference of style - either way is technically correct, but to me having both returns in the if/else block

* makes it easier to read
* makes it easier to add more elif blocks in the future

So I prefer it as is